src/components/forms/passwordForm.jsx

```diff
@@ -0,0 +1,142 @@
+      this.setState({ errors });
+    }
+  }
+
+  onChange = (event) => {
+    this.setState({
+      data: {
+        ...this.state.data,
+        [event.target.name]: event.target.value
+      }
+    });
+  };
+
+  handleBlur = ({ target: input }) => {
+    const data = { ...this.state.data };
+    const errors = { ...this.state.errors };
+    const field = input.id;
+    const newPassword = 'newPassword';
+    if (!passwordFormSchema[newPassword].test(data[newPassword])) {
+      errors.newPassword = validationMessages.newPassword;
+    }
+    if (data.confirmPassword && data.newPassword !== data.confirmPassword) {
+      errors.confirmPassword = validationMessages.confirmPassword;
+    } else if (data.confirmPassword && data.newPassword === data.confirmPassword) {
+      errors.confirmPassword = '';
+    }
+    if (!data[field]) errors[field] = `${field} is required`;
+
+    this.setState({ errors });
+  };
+
+  handleSubmit = () => {
+    const { updateUserPassword } = this.props;
+    const { currentPassword, newPassword, confirmPassword } = this.state.data;
+    updateUserPassword({ currentPassword, newPassword, confirmPassword });
+  };
+
+  render() {
+    const {
+      data: { currentPassword, newPassword, confirmPassword },
+      errors
+    } = this.state;
+
+    return (
+      <div className="col-md-6 password-form">
+        <h3 className="text-center">Change Password</h3>
+        <div>
+          <Input
+            type="password"
+            margin=".2rem"
+            id="currentPassword"
+            name="currentPassword"
+            value={currentPassword || ''}
+            width="100%"
+            wrapper="input-div"
+            handleChange={this.onChange}
+            placeholder="Current Password"
+            errorMessage={errors.currentPassword || ''}
+            onBlur={this.handleBlur}
+          />
+        </div>
+        <div>
+          <Input
+            type="password"
+            margin=".2rem"
+            name="newPassword"
+            width="100%"
+            id="newPassword"
+            value={newPassword || ''}
+            wrapper="input-div"
+            handleChange={this.onChange}
+            placeholder="New Password"
+            errorMessage={errors.newPassword || ''}
+            onBlur={this.handleBlur}
+          />
+        </div>
+        <div className="mb-30">
+          <Input
+            type="password"
+            margin=".2rem"
+            width="100%"
+            name="confirmPassword"
+            id="confirmPassword"
+            value={confirmPassword || ''}
+            wrapper="input-div mb-30"
+            handleChange={this.onChange}
+            placeholder="Confirm Password"
+            onBlur={this.handleBlur}
+            errorMessage={errors.confirmPassword}
+          />
+        </div>
+        <Button
+          sm
+          onClick={this.handleSubmit}
+          disabled={!!(!currentPassword || !newPassword || !confirmPassword)}>
```

Comment:
What exactly are you saying by this logic? 🤷‍♂️

Reply:
What I am saying is, for the button to be enabled in the password form. All fields must have been filled.

Reply:
This is less cryptic sir:
```suggestion
          disabled={!(currentPassword && newPassword && confirmPassword)}>
```